radix-engine-tests/tests/validator.rs

```diff
@@ -125,3 +126,120 @@ fn check_if_validator_accepts_delegated_stake() {
+        ManifestBuilder::new()
+            .lock_fee_from_faucet()
+            .call_method(
+                validator_address,
+                VALIDATOR_GET_REDEMPTION_VALUE_IDENT,
+                manifest_args!(Decimal::MAX),
+            )
+            .build(),
+        vec![],
+    );
+
+    // Assert
+    receipt.expect_specific_failure(|e| {
+        matches!(
+            e,
+            RuntimeError::ApplicationError(ApplicationError::ValidatorError(
+                ValidatorError::InvalidGetRedemptionAmount
+            ))
+        )
+    });
+}
+
+#[test]
+fn calling_get_redemption_value_on_staked_validator_with_smallest_amount_should_not_crash() {
+    // Arrange
+    let mut test_runner = TestRunnerBuilder::new().build();
+    let (pub_key, _, account) = test_runner.new_allocated_account();
+    let validator_address = test_runner.new_staked_validator_with_pub_key(pub_key, account);
+
+    // Act
+    let receipt = test_runner.execute_manifest(
+        ManifestBuilder::new()
+            .lock_fee_from_faucet()
+            .call_method(
+                validator_address,
+                VALIDATOR_GET_REDEMPTION_VALUE_IDENT,
+                manifest_args!(Decimal(I192::ONE)),
+            )
+            .build(),
+        vec![],
+    );
+
+    // Assert
+    receipt.expect_commit_success();
+}
+
+#[test]
+fn calling_get_redemption_value_on_staked_validator_with_min_amount_should_not_crash() {
+    // Arrange
+    let mut test_runner = TestRunnerBuilder::new().build();
+    let (pub_key, _, account) = test_runner.new_allocated_account();
+    let validator_address = test_runner.new_staked_validator_with_pub_key(pub_key, account);
+
+    // Act
+    let receipt = test_runner.execute_manifest(
+        ManifestBuilder::new()
+            .lock_fee_from_faucet()
+            .call_method(
+                validator_address,
+                VALIDATOR_GET_REDEMPTION_VALUE_IDENT,
+                manifest_args!(Decimal::MIN),
+            )
+            .build(),
+        vec![],
+    );
+
+    // Assert
+    receipt.expect_specific_failure(|e| {
+        matches!(
+            e,
+            RuntimeError::ApplicationError(ApplicationError::ValidatorError(
+                ValidatorError::InvalidGetRedemptionAmount
+            ))
+        )
+    });
+}
+
+#[test]
+fn calling_get_redemption_value_on_staked_validator_with_zero_amount_should_not_crash() {
+    // Arrange
+    let mut test_runner = TestRunnerBuilder::new().build();
+    let (pub_key, _, account) = test_runner.new_allocated_account();
+    let validator_address = test_runner.new_staked_validator_with_pub_key(pub_key, account);
+
+    // Act
+    let receipt = test_runner.execute_manifest(
+        ManifestBuilder::new()
+            .lock_fee_from_faucet()
+            .call_method(
+                validator_address,
+                VALIDATOR_GET_REDEMPTION_VALUE_IDENT,
+                manifest_args!(Decimal::MIN),
```

Comment:
I think this should be `Decimal::ZERO`? :)

Reply:
good catch!